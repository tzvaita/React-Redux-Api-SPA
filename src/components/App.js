import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
// import '../styles/App.css';
import pokeList from '../containers/pokeList';
import pokeMon from '../containers/pokeMon';

function App() {
  return (
    <div className="App">
      <div className="pokLogo">
        <a href="/">
          <img src="https://raw.githubusercontent.com/PokeAPI/media/master/logo/pokeapi_256.png" alt="POKEMON" />
        </a>
      </div>
      <Switch>
        <Route path="/" exact component={pokeList} />
        <Route path="/pokemon/:pokemon" exact component={pokeMon} />
        <Redirect to="/" />
      </Switch>
    </div>
  );
}

export default App;
