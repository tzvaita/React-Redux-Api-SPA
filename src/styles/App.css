.App {
  text-align: center;
  width: 80%;
  margin: 0 auto;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.pokLogo {
  background-color: black;
}

.singleImg {
  background-color: cadetblue;
  border: 2px solid white;
  border-radius: 15%;
  margin-top: 2px;
}

.card-title {
  text-align: left;
  text-transform: capitalize;
}

.pokTitle {
  text-decoration: none;
  color: black;
}