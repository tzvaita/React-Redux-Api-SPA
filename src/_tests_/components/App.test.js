import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import App from '../../components/App';
import '../../setupTest';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

beforeEach(() => {
  store.clearActions();
});

describe('<App />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<App />);
      // const component = wrapper.dive();
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
