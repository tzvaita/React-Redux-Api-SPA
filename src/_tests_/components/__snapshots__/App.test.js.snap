// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render() renders the component 1`] = `
<div
  className="App"
>
  <div
    className="pokLogo"
  >
    <a
      href="/"
    >
      <img
        alt="POKEMON"
        src="https://raw.githubusercontent.com/PokeAPI/media/master/logo/pokeapi_256.png"
      />
    </a>
  </div>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/pokemon/:pokemon"
    />
    <Redirect
      to="/"
    />
  </Switch>
</div>
`;
